agent/structs/config_entry_gateways.go

```diff
@@ -122,3 +122,114 @@ func (e *IngressGatewayConfigEntry) GetEnterpriseMeta() *EnterpriseMeta {
 
 	return &e.EnterpriseMeta
 }
+
+// TerminatingGatewayConfigEntry manages the configuration for a terminating service
+// with the given name.
+type TerminatingGatewayConfigEntry struct {
+	Kind     string
+	Name     string
+	Services []LinkedService
+
+	EnterpriseMeta `hcl:",squash" mapstructure:",squash"`
+	RaftIndex
+}
+
+// A LinkedService is a service represented by a terminating gateway
+type LinkedService struct {
+	// Name is the name of the service, as defined in Consul's catalog
+	Name string `json:",omitempty"`
+
+	// CAFile is the optional path to a CA certificate to use for TLS connections
+	// from the gateway to the linked service
+	CAFile string `json:",omitempty"`
+
+	// CertFile is the optional path to a client certificate to use for TLS connections
+	// from the gateway to the linked service
+	CertFile string `json:",omitempty"`
+
+	// KeyFile is the optional path to a private key to use for TLS connections
+	// from the gateway to the linked service
+	KeyFile string `json:",omitempty"`
+
+	EnterpriseMeta `hcl:",squash" mapstructure:",squash"`
+}
+
+func (e *TerminatingGatewayConfigEntry) GetKind() string {
+	return TerminatingGateway
+}
+
+func (e *TerminatingGatewayConfigEntry) GetName() string {
+	if e == nil {
+		return ""
+	}
+
+	return e.Name
+}
+
+func (e *TerminatingGatewayConfigEntry) Normalize() error {
+	if e == nil {
+		return fmt.Errorf("config entry is nil")
+	}
+
+	e.Kind = TerminatingGateway
+
+	for _, svc := range e.Services {
+		svc.EnterpriseMeta.Normalize()
```

Comment:
Realized we did not do this for ingress gateways, so thanks! I think here you would need to use a for-loop like:
```
		for i := range e.Services {
			e.Services[i].EnterpriseMeta.Normalize()
		}
```

Because `Normalize` is going to modify the struct in-place, and this range will copy the `LinkedService` value, since we are not using pointers here.

I tried to think about how to test this case and realized this is not possible to test in OSS sadly.

Reply:
Ah, good catch. Thanks!